### PR TITLE
Allow Mapbox or MapLibre from SPM on iOS, and different libs on android

### LINKED ISSUE #1273: Switching to MapLibre?
Hello community!

I recently implemented a maps feature and found out that Mapbox switched to a proprietary license with Mapbox 10. Afaik you are bound to there TOS now and have to pay their cloud fees even if you do not use them.

A lot of famous people grouped and founded the MapLibre project. Their gl-native fork can be found here: https://github.com/maplibre/maplibre-gl-native

I will soon use react-native-mapbox-gl in an upcoming project. That means we need to have a plan what happens if this repository bumps the Mapbox version to 10. What do the main contributors of this repository think about this? Is this library focused on people who use the Mapbox cloud products?

### PR DESCRIPTION
**Pull request by mfazekas**
## Description

This PR allows to specify the Mapbox library to be consumed via SwiftPackageManager. This means MapLibre for example can be used instead of Mapbox.

Fixes #1273 

Podfile:
```ruby

$RNMBGL_Use_SPM = {
  url: "https://github.com/maplibre/maplibre-gl-native-distribution",
  requirement: {
    kind: "upToNextMajorVersion",
    minimumVersion: "5.11.0"
  },
  product_name: "Mapbox"
}


pre_install do |installer|
  $RNMBGL.pre_install(installer)
end
...

post_install do |installer|
   flipper_post_install(installer)
   $RNMBGL.post_install(installer)
end
```

Top level build.gradle:
```groovy
buildscript {
    ext {
   ...
         rnmbglMapboxLibs = {
             implementation ("org.maplibre.gl:android-sdk:9.2.1")
             implementation ("com.mapbox.mapboxsdk:mapbox-sdk-turf:5.3.0")
         }

         rnmbglMapboxPlugins = {
             implementation ("com.mapbox.mapboxsdk:mapbox-android-gestures:0.7.0")
             implementation ("com.mapbox.mapboxsdk:mapbox-android-plugin-localization-v9:0.12.0")    {
                 exclude group: 'com.mapbox.mapboxsdk', module: 'mapbox-android-sdk'
             }
             implementation ("com.mapbox.mapboxsdk:mapbox-android-plugin-annotation-v9:0.8.0")        {
                 exclude group: 'com.mapbox.mapboxsdk', module: 'mapbox-android-sdk'
             }
             implementation ("com.mapbox.mapboxsdk:mapbox-android-plugin-markerview-v9:0.4.0") {
                 exclude group: 'com.mapbox.mapboxsdk', module: 'mapbox-android-sdk'
              }
         }
}
```

### CODE REVIEW

File: android/rctmgl/build.gradle [hunk 1]
```diff
@@ -36,8 +36,14 @@ dependencies {
 
     // Mapbox SDK
     if (rootProject.ext.has('rnmbglMapboxLibs')) {
-        rootProject.ext.get('rnmbglMapboxLibs').split(';').each {
-           implementation it
+        def libs = rootProject.ext.get('rnmbglMapboxLibs')
```
Comment (ferdicus) [inline android/rctmgl/build.gradle:39]: Can't we move `rnmbglMapboxLibs` and `rnmbglMapboxPlugins` in some form of constant, you know have one place to change it. 

Also, is this the recommended way to have variable dependencies? 
Not gonna complain, just feels kinda hacky :D

File: react-native-mapbox-gl.podspec [hunk 2]
```diff
@@ -12,8 +12,62 @@ TargetsToChangeToDynamic = ['MapboxMobileEvents']
...
-  # Noop
+  if $RNMBGL_Use_SPM
+    rpm_spec = {
+      url: "https://github.com/maplibre/maplibre-gl-native-distribution",
+      requirement: {
+        kind: "upToNextMajorVersion",
+        minimumVersion: "5.11.0"
+      },
+      product_name: "Mapbox"
+    }
```
Comment (ferdicus) [inline react-native-mapbox-gl.podspec:43]: Why do we need this here, when we have it in the `Podfile`, or why have it there if it's here? 
I'd assume you want to extract it from the `Podfile` to make it more user editable?

**Review by ferdicus — COMMENTED** [review]
Thanks

File: android/rctmgl/build.gradle (hunk 1 above)
Comment (mfazekas) [reply android/rctmgl/build.gradle:39]: @ferdicus sorry I'm not really familiar with groovy and/or gradle. This is what I could came up with. I've seen `rootProject.ext.get`/`rootProject.ext.has` in other projects ([example](https://github.com/callstack/react-native-pager-view/blob/master/android/build.gradle))

File: react-native-mapbox-gl.podspec (hunk 2 above)
Comment (mfazekas) [reply react-native-mapbox-gl.podspec:43]: Currently we also support `$RNMBGL_Use_SPM = true` to use Maplibre from spm with those default params. Not sure if that's desired/or good idea.

Comment (ferdicus) [reply react-native-mapbox-gl.podspec:43]: Maybe we should be opinionated towards one?
Configuration is cool and all, however I feel like the install steps are already confusing to some user.

WDYT?

**Comment by ferdicus** [comment]
@mfazekas , also let me know, when you're done with this - I can assist with the README/ Installation guides/ Changelog and test a bit

File: react-native-mapbox-gl.podspec (hunk 2 above)
Comment (mfazekas) [reply react-native-mapbox-gl.podspec:43]: For the short term I think this is for advanced users. 

Later we can consider defaulting to MapLibre as It doesn't require the access token dance.

**Comment by mfazekas** [comment]
@ferdicus I'm done with the PR

**Comment by ferdicus** [comment]
> @ferdicus I'm done with the PR

Cool, thanks for this - I'll try to test this in the coming days. 👍🏿

**Comment by jaltin** [comment]
Amazing, thanks for your enormous work on this @mfazekas and @ferdicus!

**Pull request merged**